### PR TITLE
Add read-only Releases API with Assets

### PR DESCRIPTION
**Pull request by bweir**
Hi guys,

I was using PyGithub to build a website yesterday, and found it was missing the release API, so I added it.

I later noticed that there seemed to already be an in-progress release API but it looked incomplete and didn't match the current API, so I kept going with it.

I also started on the Upload Asset function and discovered that the main API URL is hard-coded into Requester.py, so it'd take a lot more to get that feature to work.

Oh, I would have added to the development branch but there isn't one. >.>

Anyway, lemme know what you think.

### CODE REVIEW

File: github/ReleaseAsset.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+# ##############################################################################
+
+import github.GithubObject
+
+import github.NamedUser
+
+
+class ReleaseAsset(github.GithubObject.CompletableGithubObject):
+    """
+    This class represents Release Assets as returned for example by http://developer.github.com/v3/todo
```
Comment (jzelinskie) [inline github/ReleaseAsset.py]: This URL goes no where

File: github/Release.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+
+import github.GithubObject
+
+import github.NamedUser
+import github.ReleaseAsset
+
+
+class Release(github.GithubObject.CompletableGithubObject):
+    """
+    This class represents Releases as returned for example by http://developer.github.com/v3/todo
```
Comment (jzelinskie) [inline github/Release.py]: This URL goes no where

**Comment by jzelinskie** [comment]
Commits look pretty good! You should get some tests in before I merge, though.

**Comment by stleon** [comment]
Waiting releases api 👍🏻

**Comment by mfonville** [comment]
@jzelinskie if you could give me some guidance how to write the tests, I would be willing to do the last work on this PR, to get this useful work upstream

**Comment by jzelinskie** [comment]
Creating tests are documented in the `CONTRIBUTING.md` file: https://github.com/PyGithub/PyGithub/blob/master/CONTRIBUTING.md#automated-tests

**Comment by sfdye** [comment]
Close in favor of #525

**Pull request closed without merging**